### PR TITLE
[Spring]validate path exist before all steps

### PR DESCRIPTION
**Pull request by Netyyyy**
Verify that `artifact-path` and `source-path` exist before all steps in `az spring app deploy`, `az spring app deployment create` and `az spring job deploy` commands.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

File: src/spring/azext_spring/_app_validator.py [hunk 1]
```diff
@@ -162,3 +165,10 @@ def _validate_container_registry(cmd, namespace):
                 raise InvalidArgumentValueError(
                     "The instance without build service can only use '--container-image' to deploy."
                     " See more details in https://learn.microsoft.com/en-us/azure/spring-apps/how-to-deploy-with-custom-container-image?tabs=azure-cli")
+
+
+def validate_path_exist(source_path, artifact_path):
+    if source_path and not os.path.exists(source_path):
+        raise InvalidArgumentValueError('source path does not exist.')
```
Comment (yuwzho) [inline src/spring/azext_spring/_app_validator.py:172]:
```suggestion
        raise InvalidArgumentValueError('source path {} does not exist.'.format(source_path))
```

File: src/spring/azext_spring/_app_validator.py [hunk 2]
```diff
@@ -162,3 +165,10 @@ def _validate_container_registry(cmd, namespace):
...
+    if artifact_path and not os.path.exists(artifact_path):
+        raise InvalidArgumentValueError('artifact path does not exist.')
```
Comment (yuwzho) [inline src/spring/azext_spring/_app_validator.py:174]:
```suggestion
        raise InvalidArgumentValueError('artifact path {} does not exist.'.format(artifact_path))
```

**Review by yuwzho — APPROVED** [review]

**Comment by yonzhan** [comment]
Please fix CI issues.

**Comment by Netyyyy** [comment]
> Please fix CI issues.

Fixed

**Comment by Netyyyy** [comment]
@zhoxing-ms @yonzhan @yanzhudd  , please help take a review, and The failed CI is warning about the original code. Let us keep this PR clean for the new feature and we commit the incremental code passed all CI check.

File: src/spring/HISTORY.md [hunk 3]
```diff
@@ -1,5 +1,9 @@
 Release History
 ===============
+1.24.5
+---
+* Verify that `artifact-path` and `source-path` exist before all steps in `az spring app deploy`, `az spring app deployment create` and `az spring job deploy` commands.
```
Comment (yanzhudd) [inline src/spring/HISTORY.md:5]:
```suggestion
* Verify that `--artifact-path` and `--source-path` exist before all steps in `az spring app deploy`, `az spring app deployment create` and `az spring job deploy` commands.
```

**Comment by Netyyyy** [comment]
@yanzhudd do not know why pipeline failed after updating [HISTORY.md](https://github.com/Azure/azure-cli-extensions/pull/7725/files/1938c8014415257d02cd96ae6e99692068a1f507#diff-47123a6632ba9cf6cccebc15a2b1e94d988dee11c44f5513af849e0ac6ef943b) and do not have a re-run permission, please help take a look

**Comment by Netyyyy** [comment]
Hi @yanzhudd please help approve and merge, thanks

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**